mlrun/projects/project.py

```diff
@@ -2594,6 +2594,8 @@ def run(
 
         if not name and not workflow_path and not workflow_handler:
             raise ValueError("Workflow name, path, or handler must be specified")
+        if name not in self.spec._workflows.keys():
```

Comment:
should be after line 2600 since it is valid to specify a workflow name that doesn't exist along side a workflow path or callable handler.
~~~
if workflow_path or (workflow_handler and callable(workflow_handler)):
~~~

Reply:
Good thinking, Alon! I changed the position of my check. 

When you can, please see if it makes more sense now.